src/Service/FileService.php

```diff
@@ -7,91 +7,34 @@
 
 class FileService
 {
-
-
-    /**@var Registry */
+    /** @var Registry */
     private $doctrine;
-
-    private $uploadFolder;
-    /**@var StorageManager */
+    /** @var StorageManager */
     private $storageManager;
-
-    /**@var integer */
-    private $uploadMinimuNumberOfReplications;
-
-    /**@var Processor */
+    /** @var int */
+    private $uploadMinimumNumberOfReplications;
+    /** @var Processor */
     private $processor;
 
-    /**
-     * @param array{version?:string,credentials?:array{key:string,secret:string},region?:string}|null $s3Credentials
-     */
-    public function __construct(Registry $doctrine, StorageManager $storageManager, Processor $processor, string $projectDir, string $uploadFolder, string $storageFolder, bool $createDbStorageService, string $elasticmsRemoteServer, string $elasticmsRemoteAuthkey, string $sftpServer, string $sftpPath, string $sftpUser, string $publicKey, string $privateKey, array $s3Credentials = null, string $s3Bucket = null)
+    public function __construct(Registry $doctrine, StorageManager $storageManager, Processor $processor)
     {
         $this->doctrine = $doctrine;
-        $this->uploadFolder = $uploadFolder;
         $this->storageManager = $storageManager;
         $this->processor = $processor;
-        $this->uploadMinimuNumberOfReplications = 10;
-
-        if ($storageFolder && !empty($storageFolder)) {
-            if (substr($storageFolder, 0, 2) === ('.' . DIRECTORY_SEPARATOR)) {
-                $this->addStorageService(new FileSystemStorage($projectDir . substr($storageFolder, 1)));
-            } else {
-                $this->addStorageService(new FileSystemStorage($storageFolder));
-            }
-        }
-
-        if (!empty($s3Credentials) && !empty($s3Bucket)) {
-            $this->addStorageService(new S3Storage($s3Credentials, $s3Bucket));
-        }
-
-
-        if ($createDbStorageService) {
-            $this->addStorageService(new EntityStorage($doctrine));
-        }
-
-        if (!empty($elasticmsRemoteServer)) {
-            $this->addStorageService(new HttpStorage($elasticmsRemoteServer, '/data/file/view/', $elasticmsRemoteAuthkey));
-        }
-
-        if (!empty($sftpServer) && !empty($sftpPath) && !empty($sftpPath) && !empty($publicKey) && !empty($privateKey)) {
-            $this->addStorageService(new SftpStorage($sftpServer, $sftpPath, $sftpUser, $publicKey, $privateKey, true));
-        }
-    }
-
-    public function addStorageService(StorageInterface $storageAdapter): void
-    {
-        $this->storageManager->addAdapter($storageAdapter);
-    }
-
-
-    public function getStorageService(string $storageServiceId): ?StorageInterface
-    {
-        foreach ($this->storageManager->getAdapters() as $storageService) {
-            if ($storageService->__toString() == $storageServiceId) {
-                return $storageService;
-            }
-        };
-        return null;
+        $this->uploadMinimumNumberOfReplications = 10;
```

Comment:
This is not new code, but should we not have this in config?
```suggestion
        $this->uploadMinimumNumberOfReplications = 10;
```

Reply:
I thinking of adding a `read-only` and a `to-skip` flag in the EMS_STORAGE and implement a isReadOnly and  isToSkip methods in the storage services?